### ISSUE TITLE
[Accessibility]: Saved searches screen reader issues

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#35162]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34658

### Problem description

- [ ] Nothing announced when navigating the Saved Searches page. Expected: the "Saved searches" heading should be focused.
- [ ] Nothing announced/focused when clicking "+ Add saved search" button or clicking the "settings" button on a saved search. Expected: Heading ("Add saved search"/"Managed saved search") should be focused.
- [ ] Nothing announced when submitting the form to add a saved search. Expected: confirmation read and focused is returned to either the "+ Add saved search" button or the newly created saved search in the list.
- [ ] On create/edit page, on the bottom link to code moniroting, the right arrow is read by the screen reader. Expected: the arrow should be aria-hidden  <img width="223" alt="image" src="https://user-images.githubusercontent.com/206864/167490904-9796120d-b119-4b71-958d-957ef5a77ca0.png">
- [ ] On list of saved searches, it's confusing what clicking on the saved search name will do. Expcted: the saved search name link should have a visually-hidden component that says "Click to run saved search" or similar.
- [ ] On list of saved searches, after deleting a saved search, no confirmation is read. Expected: Announcement should be read confirming the deletion.

### Expected behavior

Issues are resolved as described above.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by GitStart. I have assigned the label 'gitstart' and added the assignee 'gitstart-sourcegraph'.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.

**Comment by gitstart-sourcegraph** [comment]
Hi @limitedmage 
With regards to this requirement: `Nothing announced when navigating the Saved Searches page. Expected: the "Saved searches" heading should be focused.`,
We were able to announce `Saved Searches page` navigation by using the `PageTitle` component, which resolves a fraction of this requirement. 
However  for `heading should be focused`, want to confirm if this requirement was to facilitate announcement of the page heading i.e since we are able to announce the page title without focusing on the header text, would it still be beneficial to trigger focus on the header text after navigation.

**Comment by limitedmage** [comment]
I think with the page title changes, the "focus" part of this is not necessary anymore (I created this issue before the work in page title was done). Thank you!

**Comment by gitstart-sourcegraph** [comment]
@limitedmage Thanks for the feedback. Could you help to confirm that your feedback included `confirmation read and focused is returned to either the "+ Add saved search" button or the newly created saved search in the list`
